packages/cozy-client/src/models/konnectorFolder.js

```diff
@@ -244,3 +281,157 @@ export const buildFolderPermission = folder => {
+ */
+const hasKonnectorReference = (folder, slug) =>
+  Boolean(
+    folder.referenced_by?.find(
+      ref =>
+        ref.type === 'io.cozy.konnectors' &&
+        ref.id === 'io.cozy.konnectors/' + slug
+    )
+  )
+
+/**
+ * Test if a given folder has a given source account identifier reference
+ *
+ * @param {import('../types').IOCozyFolder} folder - folder to test
+ * @param {string} sourceAccountIdentifier - source account identifier
+ * @returns {boolean}
+ */
+const hasSourceAccountIdentifierReference = (
+  folder,
+  sourceAccountIdentifier
+) => {
+  return Boolean(
+    folder.referenced_by?.find(
+      ref =>
+        ref.type === 'io.cozy.accounts.sourceAccountIdentifier' &&
+        ref.id === sourceAccountIdentifier
+    )
+  )
+}
+
+/**
+ * Ensure that a given folder has the given konnector reference
+ *
+ * @param {object} options - options object
+ * @param {CozyClient} options.client - CozyClient instance
+ * @param {import('../types').IOCozyFolder} options.folder - folder to test
+ * @param {import('../types').IOCozyKonnector} options.konnector - konnector to which we want the reference
+ * @returns {Promise<void>}
+ */
+const ensureKonnectorReference = async ({ client, folder, konnector }) => {
+  const fileCollection = client.collection('io.cozy.files')
+  if (!hasKonnectorReference(folder, konnector.slug)) {
+    await fileCollection.addReferencesTo(konnector, [folder])
+  }
+}
+
+/**
+ * Ensure that a given folder has the given source account identifier reference
+ *
+ * @param {object} options - options object
+ * @param {CozyClient} options.client - CozyClient instance
+ * @param {import('../types').IOCozyFolder} options.folder - folder to test
+ * @param {string} options.sourceAccountIdentifier - source account identifier to which we want the reference
+ * @returns {Promise<void>}
+ */
+const ensureSourceAccountIdentifierReference = async ({
+  client,
+  folder,
+  sourceAccountIdentifier
+}) => {
+  const fileCollection = client.collection('io.cozy.files')
+  if (!hasSourceAccountIdentifierReference(folder, sourceAccountIdentifier)) {
+    await fileCollection.addReferencesTo(
+      {
+        _id: sourceAccountIdentifier,
+        _type: 'io.cozy.accounts.sourceAccountIdentifier'
+      },
+      [folder]
+    )
+  }
+}
+
+/**
+ * try to find a konnector account folder using file references
+ *
+ * @param {object} options - options object
+ * @param {CozyClient} options.client - CozyClient instance
+ * @param {string} options.slug - konnector slug
+ * @param {string} options.sourceAccountIdentifier - source account identifier
+ * @returns {Promise<import('../types').IOCozyFolder>}
+ */
+export const findKonnectorAccountFolderByReference = async ({
+  client,
+  slug,
+  sourceAccountIdentifier
+}) => {
+  const fileCollection = client.collection('io.cozy.files')
+  const resp = await fileCollection.findReferencedBy({
+    _type: 'io.cozy.konnectors',
+    _id: `io.cozy.konnectors/${slug}`
+  })
```

Comment:
Why not using  cozy-client query ? e.g. `client.query(Q('io.cozy.files').referencedBy({...})`
This way, you will benefit from the store